### PR TITLE
Get rid of "stale 'now' ", In case IdentityModel ConfigurationManager takes long time to get Idp Metadata

### PR DESCRIPTION
**Pull request by RyanLiu99**

### CODE REVIEW

**Comment by RyanLiu99** [comment]
This is to address "stale now" issue.

This meta data is cached for x seconds to reduce the call to Idp server to get meta data.  But to make the cache works, a global lock is imposed cross requests.  Cache start time is set before calling to Id server, before waiting for the lock, not after the call returns. I call it a “stale now”. This is  a problem when call to Idp server take longer then x seconds.  Ocz x seconds has been changes from 30s to 5m from version 6.8 to version 6.27, make it less sever.

When the call to IdentiyProvider has issue, take much longer time, all requests are blocked at this lock. 

Once the call eventually returned from Idp server, SOME requests piling at the lock can just take cached data and  keep going. But if the call to Id server take longer than cache time x seconds, those requests come later than stale now + x seconds, still need call Id server again even there is fresh cached data. The global lock is blocking 2nd half requests and new requests again.  And also impose more stress on already stressed Idp server since it does not wait x seconds and call it right way again. This can get worse and worse.

File: src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs [hunk 1]
```diff
@@ -127,16 +127,16 @@ public async Task<T> GetConfigurationAsync(CancellationToken cancel)
             await _refreshLock.WaitAsync(cancel).ConfigureAwait(false);
             try
             {
-                if (_syncAfter <= now)
+                if (_syncAfter <= DateTimeOffset.UtcNow)
```
Comment (brentschmaltz) [inline src/Microsoft.IdentityModel.Protocols/Configuration/ConfigurationManager.cs:162]: @RyanLiu99 can you removed the local variable "now">

**Review by brentschmaltz — CHANGES_REQUESTED** [review]
nit: remove local variable: 'now'.

**Review by brentschmaltz — APPROVED** [review]

**Comment by brentschmaltz** [comment]
@RyanLiu99 for some reason this is showing up with a conflict, can you rebase on top of dev.

**Comment by RyanLiu99** [comment]
@brentschmaltz   I just merged dev branch into my RyanLiu-patch-1, no conflicts at all. I also merged up-to-date dev branch into RyanLiu-patch-1 in my folk, no issues either.

 Can you try again? If there still issue, I will rebase.

**Review by brentschmaltz — APPROVED** [review]

**Comment by brentschmaltz** [comment]
@RyanLiu99 something odd is going on, git is blocking this PR.
Perhaps start with a new branch?
Last resort, i can do a manual merge, but would like to see your name on the PR.

**Comment by RyanLiu99** [comment]
Ok, @brentschmaltz  I'll do it .

**Comment by RyanLiu99** [comment]
A new PR created.  I will close this one.  @brentschmaltz 
https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/pull/2063
Sorry take a while, I was travelling.

**Pull request closed without merging**